test-suite-data.json

```
@@ -466,5 +466,77 @@
+    "type": "huggingface",
+    "namespace": "microsoft",
+    "name": "deberta-v3-base",
+    "version": "559062ad13d311b87b2c455e67dcd5f1c8f65111",
+    "qualifiers": {"repository_url": "https://hub-ci.huggingface.co"},
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "Hugging Face model with various cases",
+    "purl": "pkg:huggingface/EleutherAI/gpt-neo-1.3B@797174552AE47F449AB70B684CABCB6603E5E85E",
+    "canonical_purl": "pkg:huggingface/EleutherAI/gpt-neo-1.3B@797174552ae47f449ab70b684cabcb6603e5e85e",
+    "type": "huggingface",
+    "namespace": "EleutherAI",
+    "name": "gpt-neo-1.3B",
+    "version": "797174552ae47f449ab70b684cabcb6603e5e85e",
+    "qualifiers": null,
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "MLflow model tracked in Azure Databricks (case insensitive)",
+    "purl": "pkg:mlflow/CreditFraud@3?repository_url=https://adb-5245952564735461.0.azuredatabricks.net/api/2.0/mlflow",
+    "canonical_purl": "pkg:mlflow/creditfraud@3?repository_url=https://adb-5245952564735461.0.azuredatabricks.net/api/2.0/mlflow",
+    "type": "mlflow",
+    "namespace": null,
+    "name": "creditfraud",
+    "version": "3",
+    "qualifiers": {"repository_url": "https://adb-5245952564735461.0.azuredatabricks.net/api/2.0/mlflow"},
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "MLflow model tracked in Azure ML (case sensitive)",
+    "purl": "pkg:mlflow/CreditFraud@3?repository_url=https://westus2.api.azureml.ms/mlflow/v1.0/subscriptions/a50f2011-fab8-4164-af23-c62881ef8c95/resourceGroups/TestResourceGroup/providers/Microsoft.MachineLearningServices/workspaces/TestWorkspace",
+    "canonical_purl": "pkg:mlflow/CreditFraud@3?repository_url=https://westus2.api.azureml.ms/mlflow/v1.0/subscriptions/a50f2011-fab8-4164-af23-c62881ef8c95/resourceGroups/TestResourceGroup/providers/Microsoft.MachineLearningServices/workspaces/TestWorkspace",
+    "type": "mlflow",
+    "namespace": null,
+    "name": "CreditFraud",
+    "version": "3",
+    "qualifiers": {"repository_url": "https://westus2.api.azureml.ms/mlflow/v1.0/subscriptions/a50f2011-fab8-4164-af23-c62881ef8c95/resourceGroups/TestResourceGroup/providers/Microsoft.MachineLearningServices/workspaces/TestWorkspace"},
+    "subpath": null,
+    "is_invalid": false
+  },
+  {
+    "description": "MLflow model with unique identifiers",
+    "purl": "pkg:mlflow/trafficsigns@10?model_uuid=36233173b22f4c89b451f1228d700d49&run_id=410a3121-2709-4f88-98dd-dba0ef056b0a&repository_url=https://adb-5245952564735461.0.azuredatabricks.net/api/2.0/mlflow",
+    "canonical_purl": "pkg:mlflow/trafficsigns@10?model_uuid=36233173b22f4c89b451f1228d700d49&run_id=410a3121-2709-4f88-98dd-dba0ef056b0a&repository_url=https://adb-5245952564735461.0.azuredatabricks.net/api/2.0/mlflow",
```

I believe the ordering of the qualifiers is invalid here, no? Running into some test failures.

I'd expect this to be like so:
```suggestion
    "canonical_purl": "pkg:mlflow/trafficsigns@10?model_uuid=36233173b22f4c89b451f1228d700d49&repository_url=https://adb-5245952564735461.0.azuredatabricks.net/api/2.0/mlflow&run_id=410a3121-2709-4f88-98dd-dba0ef056b0a",
```